internal/gcsx/random_reader.go

```diff
@@ -256,14 +255,15 @@ func (rr *randomReader) ReadAt(
 
 	// Note: If we are reading the file for the first time and read type is sequential
 	// then the file cache behavior is write-through i.e. data is first read from
-	// GCS, cached in file and then served from that file. Also, the cacheHit is
-	// true in that case.
+	// GCS, cached in file and then served from that file. But the cacheHit is
+	// false in that case.
 	n, cacheHit, err = rr.tryReadingFromFileCache(ctx, p, offset)
 	if err != nil {
 		err = fmt.Errorf("ReadAt: while reading from cache: %v", err)
 		return
 	}
-	if cacheHit {
+	// data was served from cache.
+	if n == len(p) {
```

Comment:
kernel always send 1mib buffer, so a case where offset + 1 mib < object size, this check will fail.
So, can we keep additional check to compare like n == len(p) || (n < len(p) && offset + n == rr.object.Size) ?